cryostatio/cryostat-operator
# chore(cryostat3): various cleanups

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Related to #710

## Description of the change:
- cleans up some unused volume mounts and env vars
- removes Grafana basic auth secret - no longer needed, since Grafana is placed behind the same auth proxy as Cryostat
- configures Grafana for anonymous access (not really anonymous since traffic must go through auth proxy first)
- corrects environment variables so Cryostat UI "View in Grafana" correctly opens Grafana UI
- renames some CR fields to better reflect what they mean in 3.0
- updates docs/config.md
- removes AuthProperties CR field since this is replaced by AuthorizationOptions now

## Motivation for the change:
General clean up, get Grafana working.

## How to manually test:
1. Check out, build, and test PR. Use `OPENSHIFT_OAUTH_PROXY_IMG=quay.io/andrewazores/openshift-oauth-proxy:test-14` (rebuild with golang 1.22) so the UI is usable without the proxy path redirect bug.
2. Create Cryostat CR with cert_manager integration disabled
3. Note Application Route URL, open a new private/incognito browser tab and visit ex. `http://cryostat-cryostat.apps-crc.testing/grafana` to go directly to the Grafana dashboard. This should prompt you to log in with OpenShift, just like visiting the Cryostat UI.
4. In Cryostat UI, log in with OpenShift, then define a `localhost:0` custom target, start a recording, and "View in Grafana" it. This should pass you directly through to Grafana without re-auth.
5. Copy and paste the "View in Grafana" tab URL into a new incognito window to simulate a different, unauthenticated user. Paste and go. You should be prompted to log in.
6. Archive the recording and repeat the same "View in Grafana" tests.
